src/EspressoSGXTEEVerifier.sol

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {V3QuoteVerifier} from
+    "@automata-network/dcap-attestation/contracts/verifiers/V3QuoteVerifier.sol";
+import {BELE} from "@automata-network/dcap-attestation/contracts/utils/BELE.sol";
+import {Header} from "@automata-network/dcap-attestation/contracts/types/CommonStruct.sol";
+import {
+    HEADER_LENGTH,
+    ENCLAVE_REPORT_LENGTH
+} from "@automata-network/dcap-attestation/contracts/types/Constants.sol";
+import {EnclaveReport} from "@automata-network/dcap-attestation/contracts/types/V3Structs.sol";
+import {BytesUtils} from "@automata-network/dcap-attestation/contracts/utils/BytesUtils.sol";
+import "@openzeppelin/contracts/access/Ownable2Step.sol";
+import {IEspressoSGXTEEVerifier} from "./interface/IEspressoSGXTEEVerifier.sol";
+
+/**
+ *
+ * @title  Verifies quotes from the TEE and attests on-chain
+ * @notice Contains the logic to verify a quote from the TEE and attest on-chain. It uses the V3QuoteVerifier contract
+ *         from automata to verify the quote. Along with some additional verification logic.
+ */
+contract EspressoSGXTEEVerifier is IEspressoSGXTEEVerifier, Ownable2Step {
+    using BytesUtils for bytes;
+
+    // V3QuoteVerififer contract from automata to verify the quote
+    V3QuoteVerifier public quoteVerifier;
+
+    mapping(bytes32 => bool) public registeredEnclaveHash;
+    mapping(bytes32 => bool) public registeredEnclaveSigner;
```

Comment:
just curious, what's the difference between `registeredSigners` and `registeredEnclaveSigners`?

and I suppose the enclave signers is stored as bytes32 because it's a hash and not an ethereum address?

Reply:
So each batch poster will generate an `ephemeral key` which each TEE will attest to. We decided to do this because otherwise, we add to submit an attestation with each batch. Now the batch poster can generate an `ephemeral key` and sign the batches using that key instead of sending an attestation every time. 

EnclaveSigners is the TEE signer and registeredSigner is the ephemeral key and yes its the hash of the EnclaveSigner key thats stored

Reply:
ah got it - where can i find the best diagram/description to explain the workflow of this?